tobac/utils/internal/general_internal.py

```diff
@@ -452,12 +467,15 @@ def njit_if_available(func, **kwargs):
         return func
 
 
-def find_vertical_axis_from_coord(variable_cube, vertical_coord=None):
+def find_vertical_axis_from_coord(
+    variable_cube: Union[iris.cube.Cube, xr.DataArray],
+    vertical_coord: Union[str, None] = None,
```

Comment:
I find `|` clearer than `union` for type hints, but does that work for versions of python pre 3.10?

Reply:
I looked into this and wasn't able to come to a firm conclusion as to when the `from __future__ import annotations` allow the `|` syntax. Unfortunately, I think matrix testing needs to move up the priority list to catch stuff like this.

Reply:
#353 addresses matrix testing